### PR TITLE
Add package check

### PR DESCRIPTION
**Pull request by stramel**
Adds a package check and covers the rest of the outstanding issues. 

https://www.skypack.dev/blog/2020/10/skypack-quality-score-actionable-feedback-to-build-better-packages/
https://docs.skypack.dev/package-authors/package-checks

### CODE REVIEW

File: .github/workflows/unitTest.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ jobs:
       - uses: actions/checkout@v2
       - name: Install dependencies
         run: yarn install --frozen-lockfile
+      - name: Package Check
+        run: |
+          npm run check:package
```
Comment (drwpow) [inline .github/workflows/unitTest.yml:15]: 🙂

**Comment by bluebill1049** [comment]
Thanks @stramel

**Comment by klarstrup** [comment]
just FYI @stramel @bluebill1049 this breaks other left-out exports like `/dist/index.ie11.js` in some bundlers

**Comment by stramel** [comment]
> just FYI @stramel @bluebill1049 this breaks other left-out exports like `/dist/index.ie11.js` in some bundlers

Hmmm can you provide anymore details on the setup you're referring to?

**Comment by klarstrup** [comment]
I should've said "environment" rather than "bundler", as this also affects plain node. 

I'm not at a computer to reproduce right now but you can read about `exports` here: https://nodejs.org/api/packages.html#packages_package_entry_points

With this being a very relevant paragraph:
> Warning: Introducing the "exports" field prevents consumers of a package from using any entry points that are not defined, including the package.json (e.g. require('your-package/package.json'). This will likely be a breaking change.

**Comment by stramel** [comment]
> I should've said "environment" rather than "bundler", as this also affects plain node. 
> 
> I'm not at a computer to reproduce right now but you can read about `exports` here: https://nodejs.org/api/packages.html#packages_package_entry_points
> 
> With this being a very relevant paragraph:
> > Warning: Introducing the "exports" field prevents consumers of a package from using any entry points that are not defined, including the package.json (e.g. require('your-package/package.json'). This will likely be a breaking change.

It looks like we can define a custom environment for the ie11 and umd exports and define a good default. I will put out a pr later today and do some more testing with this in various environments. 

Thanks for bringing it to our attention

**Comment by bluebill1049** [comment]
I think this is the fix for that: https://github.com/react-hook-form/react-hook-form/pull/3274

**Pull request merged**